### ISSUE TITLE
[QH] Preserve independent child settings in compositional TargetConfig assembly

### ISSUE THREAD

**Issue by JanDuchscherer104** [JanDuchscherer104/ARIA-NBV#284]
## Outcome

Make new model composition derive compatible dimensions without overwriting independently configured stage hyperparameters.

## Scope and constraints

Preserve every existing concrete model, feature option, checkpoint/config identity and generic horizon interface. This is additive follow-up to the September 7 scorer review and its user correction; it does not select a new default or change the accepted thesis scope. External operations should use existing pinned dependencies with semantic/API checks before implementation.

- Use existing propagation_exclude_fields for independently owned dropout, hidden_dim and discriminators; do not change repository-wide propagation semantics.
- Every meaningful configurable stage, including deterministic readout, uses TargetConfig/target_type/setup_target. Internal Linear/reduction primitives need no separate factory.
- Derive feature widths/layout at construction; inject runtime dependencies explicitly. Validate spatial/summary and decoder/objective compatibility before forward without dummy GPU execution.
- Use explicit known config unions and preserve old serialized identities; no import strings, registry, DAG executor or duplicate factory. Construct once outside forward; explicitly reuse one module instance when weights are shared.

## Acceptance

- [ ] Changing fusion settings preserves independent history/readout settings through TOML round trip.
- [ ] Dimension or modality mismatch fails with the violated semantic interface.
- [ ] Factories run once; shared weights have shared object identity; registration/device/train-eval/state-dict lifecycle works.

## Verification

- cd aria_nbv && uv run pytest tests/utils -k config
- cd aria_nbv && uv run pytest tests/vin/test_target_finite_horizon.py tests/lightning

## Owners and provenance

- [aria_nbv/aria_nbv/utils/base_config.py](https://github.com/JanDuchscherer104/ARIA-NBV/blob/b6232da772c04b4c5061618b1d110771493f3505/aria_nbv/aria_nbv/utils/base_config.py)
- [aria_nbv/aria_nbv/lightning/qh_experiment.py](https://github.com/JanDuchscherer104/ARIA-NBV/blob/b6232da772c04b4c5061618b1d110771493f3505/aria_nbv/aria_nbv/lightning/qh_experiment.py)
- [aria_nbv/aria_nbv/vin/models/target_finite_horizon.py](https://github.com/JanDuchscherer104/ARIA-NBV/blob/b6232da772c04b4c5061618b1d110771493f3505/aria_nbv/aria_nbv/vin/models/target_finite_horizon.py)

Origin: September 7, 2026 QH scorer architecture review, refined by the explicit requirement to retain all model implementations. Source performance observations are historical and must be refreshed before scientific claims. Review code anchor: `b6232da772c04b4c5061618b1d110771493f3505`.

Local work mapping: `todo-097`; `issue-033`, `issue-028`.
